samcli/commands/init/init_templates.py

```diff
@@ -0,0 +1,128 @@
+        options = self.init_options(runtime, dependency_manager)
+        try:
+            template = next(item for item in options if self._check_app_template(item, app_template))
+            if template.get("init_location") is not None:
+                return template["init_location"]
+            elif template.get("directory") is not None:
+                return os.path.join(self.repo_path, template["directory"])
+            else:
+                raise UserException("Invalid template. This should not be possible, please raise an issue.")
+        except StopIteration:
+            msg = "Can't find application template " + app_template + " - check valid values in interactive init."
+            raise UserException(msg)
+
+    def _check_app_template(self, entry, app_template):
+        return entry["appTemplate"] == app_template
+
+    def init_options(self, runtime, dependency_manager):
+        if self.clone_attempted is False:
+            self._clone_repo()
+        if self.repo_path is None:
+            return self._init_options_from_bundle(runtime, dependency_manager)
+        return self._init_options_from_manifest(runtime, dependency_manager)
+
+    def _init_options_from_manifest(self, runtime, dependency_manager):
+        manifest_path = os.path.join(self.repo_path, "manifest.json")
+        with open(str(manifest_path)) as fp:
+            body = fp.read()
+            manifest_body = json.loads(body)
+            templates = manifest_body.get(runtime)
+            if templates is None:
+                # Fallback to bundled templates
+                return self._init_options_from_bundle(runtime, dependency_manager)
+            if dependency_manager is not None:
+                templates_by_dep = itertools.takewhile(lambda x: x["dependencyManager"] == dependency_manager, templates)
+                return list(templates_by_dep)
+            return templates
+
+    def _init_options_from_bundle(self, runtime, dependency_manager):
+        for mapping in list(itertools.chain(*(RUNTIME_DEP_TEMPLATE_MAPPING.values()))):
+            if runtime in mapping["runtimes"] or any([r.startswith(runtime) for r in mapping["runtimes"]]):
+                if not dependency_manager or dependency_manager == mapping["dependency_manager"]:
+                    mapping["appTemplate"] = "hello-world"  # when bundled, use this default template name
+                    return [mapping]
+        msg = "Lambda Runtime {} and dependency manager {} does not have an available initialization template.".format(
+            runtime, dependency_manager
+        )
+        raise UserException(msg)
+
+    def _clone_repo(self):
+        shared_dir = global_cfg.config_dir
+        expected_path = os.path.normpath(os.path.join(shared_dir, self._repo_name))
+        if self._should_clone_repo(expected_path):
+            try:
+                subprocess.check_output(["git", "clone", self._repo_url], cwd=shared_dir, stderr=subprocess.STDOUT)
```

Comment:
on windows you might need to look for git.exe or git.cmd, Not sure which.

Reply:
Good callout, adding an investigation item on this.

Reply:
Took a modified approach based on what we do for AWS CLI. On the one hand, we may want to abstract that to a utility, on the other hand, AWS CLI dependency is going away anyways.